cmd/crates/stellar-ledger/src/lib.rs

```diff
@@ -74,69 +73,60 @@ pub enum LedgerError {
     Bip32PathError(String),
 
     #[error(transparent)]
-    XdrError(#[from] XdrError),
+    XdrError(#[from] xdr::Error),
 
     #[error(transparent)]
     DecodeError(#[from] DecodeError),
 }
 
-pub struct LedgerOptions<T: Exchange> {
-    exchange: T,
-    hd_path: slip10::BIP32Path,
-}
-
 pub struct LedgerSigner<T: Exchange> {
-    network_passphrase: String,
     transport: T,
-    hd_path: slip10::BIP32Path,
 }
 
 impl<T> LedgerSigner<T>
 where
     T: Exchange,
 {
+    pub fn new(transport: T) -> Self {
+        Self { transport }
+    }
+    pub fn native() -> Result<LedgerSigner<TransportNativeHID>, Error> {
+        Ok(LedgerSigner {
+            transport: get_transport()?,
+        })
+    }
+}
+
+impl<T> Blob for LedgerSigner<T>
+where
+    T: Exchange,
+{
+    type Key = HdPath;
+    type Error = Error;
     /// Get the public key from the device
     /// # Errors
     /// Returns an error if there is an issue with connecting with the device or getting the public key from the device
-    pub async fn get_public_key(
+    async fn get_public_key(
         &self,
-        index: u32,
-    ) -> Result<stellar_strkey::ed25519::PublicKey, LedgerError> {
-        let hd_path = bip_path_from_index(index)?;
-        Self::get_public_key_with_display_flag(self, hd_path, false).await
-    }
-
-    /// Get the device app's configuration
-    /// # Errors
-    /// Returns an error if there is an issue with connecting with the device or getting the config from the device
-    pub async fn get_app_configuration(&self) -> Result<Vec<u8>, LedgerError> {
-        let command = APDUCommand {
-            cla: CLA,
-            ins: GET_APP_CONFIGURATION,
-            p1: P1_GET_APP_CONFIGURATION,
-            p2: P2_GET_APP_CONFIGURATION,
-            data: vec![],
-        };
-        self.send_command_to_ledger(command).await
+        index: impl Into<Self::Key>,
+    ) -> Result<stellar_strkey::ed25519::PublicKey, Error> {
+        // let hd_path = bip_path_from_index(index)?;
+        Self::get_public_key_with_display_flag(self, index.into(), false).await
```

Comment:
I'm seeing an error for both `get_public_key` and `sign_blob` (when I add `async_trait` to the Blob trait). I'm a bit stumped about the suggested solution that the compiler gives:

```
future cannot be sent between threads safely
required for the cast from `Pin<Box<{async block@cmd/crates/stellar-ledger/src/lib.rs:115:60: 117:6}>>` to `Pin<Box<(dyn Future<Output = std::result::Result<stellar_strkey::ed25519::PublicKey, Error>> + Send + 'async_trait)>>`rustc[Click for full compiler diagnostic](rust-analyzer-diagnostics-view:/diagnostic%20message%20%5B1%5D?1#file%3A%2F%2F%2FUsers%2Felizabethengelman%2FProjects%2FAha-Labs%2Fstellar%2Fsoroban-cli%2Fcmd%2Fcrates%2Fstellar-ledger%2Fsrc%2Flib.rs)
lib.rs(114, 9): captured value is not `Send`
lib.rs(114, 36): consider further restricting this bound: ` + std::marker::Send`
```